src/tools/ip/quicip.h

```diff
@@ -72,8 +98,20 @@ ClientConnectionCallback(
     _Inout_ QUIC_CONNECTION_EVENT* Event
     )
 {
-    CALLBACK_CONTEXT* Context = (CALLBACK_CONTEXT*)_Context;
+    QUIC_IP_LOOKUP* Context = (QUIC_IP_LOOKUP*)_Context;
     switch (Event->Type) {
+    case QUIC_CONNECTION_EVENT_SHUTDOWN_INITIATED_BY_TRANSPORT:
+        if (!Context->Success && Context->Status == QUIC_IP_DEFAULT_STATUS) {
+            QUIC_PRINTF("Connection Shutdown, 0x%x!\n", Event->SHUTDOWN_INITIATED_BY_TRANSPORT.Status);
+            Context->Status = Event->SHUTDOWN_INITIATED_BY_TRANSPORT.Status;
+        }
+        break;
+    case QUIC_CONNECTION_EVENT_SHUTDOWN_INITIATED_BY_PEER:
+        if (!Context->Success && Context->Status == QUIC_IP_DEFAULT_STATUS) {
+            QUIC_PRINTF("Connection Shutdown by Peer, 0x%llx!\n", Event->SHUTDOWN_INITIATED_BY_PEER.ErrorCode);
```

Comment:
Apparently, Linux doesn't like this:
```
/home/vsts/work/1/s/src/tools/ip/quicip.h:24:56: error: format ‘%llx’ expects argument of type ‘long long unsigned int’, but argument 2 has type ‘QUIC_UINT62 {aka long unsigned int}’ [-Werror=format=]
 #define QUIC_PRINTF(fmt, ...) printf(fmt, ##__VA_ARGS__)
                                                        ^
/home/vsts/work/1/s/src/tools/ip/quicip.h:73:64:
             QUIC_PRINTF("Stream Peer Send Aborted, 0x%llx!\n", Event->PEER_SEND_ABORTED.ErrorCode);
```
What's the best way forward?

Reply:
#include <inttypes.h>

Then instead of `0x%llx!` use `%"PRIx64"` (The quotes are important here)

Reply:
Just noticed because C++ you need spaces between the end quotes and the macro

Reply:
Just decided to remove them all together.